blocks/linkable-cards/linkable-cards.js

```diff
@@ -1,11 +1,23 @@
-export default function decorate(block) {
+export default async function decorate(block) {
   let html = '';
-
-  [...block.children].forEach((element) => {
-    const content = element.children[0].innerHTML;
-    const lastColumn = element.children[element.children.length - 1];
-    const link = lastColumn.querySelectorAll('a')[0].getAttribute('href');
-    html += `<div><a href="${link}">${content}</a></div>`;
-    block.innerHTML = html;
-  });
+  if ([...block.children].length > 0) {
+    [...block.children].forEach((element) => {
+      const content = element.children[0].innerHTML;
+      const lastColumn = element.children[element.children.length - 1];
+      const link = lastColumn.querySelectorAll('a')[0].getAttribute('href');
+      html += `<div><a href="${link}">${content}</a></div>`;
+      block.innerHTML = html;
+    });
+  } else { // fetchign documents and populating top 3 cards.
+    const resp = await fetch(`${window.hlx.codeBasePath}/documents/query-index.json`);
+    const json = await resp.json();
+    [...json.data].forEach((element, index) => {
+      if (index < 3) {
+        html += `<div><a href="e${element.path}"><p>${element.parent}</p>
+        <h3 id="for-contract-research-organizations">${element.title}</h3>
+        <p>${element.description}</p></a></div>`;
+        block.innerHTML = html;
+      }
+    });
+  }
```

Comment:
We'll need to move this `else` fetching method to the `Related Resources` block (https://github.com/hlxsites/merative2/blob/main/blocks/related-resources/related-resources.js) as this is what it's being used on the existing Case Study and other doc pages (see - [sharepoint doc](https://merative.sharepoint.com/:w:/r/sites/Merative.comWebsite/_layouts/15/Doc.aspx?sourcedoc=%7B4D3CE5C4-9E61-4328-BD14-293C79B7347D%7D&file=4-ways-case-study.docx&action=default&mobileredirect=true)).

The goal is to display the latest 3 posts from the document query index when this block is empty. Also, we should query from the same `assettype` value and not just grab the top 3 items from the query index (https://main--merative2--hlxsites.hlx.page/documents/query-index.json).

![Screen Shot 2023-06-22 at 10 33 58 AM](https://github.com/hlxsites/merative2/assets/1815714/c6e4e50f-261b-4f1e-ac57-061939e18751)

Reply:
Done